src/testdir/test_quickfix.vim

```diff
@@ -6578,4 +6578,32 @@ func Test_cbuffer_range()
   call XbufferTests_range('l')
 endfunc
 
+" Test for displaying fname pass from setqflist when the name
+" are hard links to prevent seemly duplicate entries.
+func Xtest_hardlink_fname(cchar)
```

Comment:
You need to add another function starting with the prefix "Test_" that calls this function with the argument 'c' and 'l'.